src/test/java/org/openrewrite/java/migrate/apache/commons/lang/IsNotEmptyToJdkTest.java

```diff
@@ -0,0 +1,124 @@
+ */
+package org.openrewrite.java.migrate.apache.commons.lang;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class IsNotEmptyToJdkTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.parser(JavaParser.fromJavaVersion().classpath("commons-lang3", "plexus-utils", "maven-shared-utils"))
+          .recipe(new IsNotEmptyToJdk());
+    }
+
+    @ParameterizedTest
+    @CsvSource(delimiter = '#', textBlock = """
+      org.apache.commons.lang3.StringUtils # StringUtils.isEmpty(first) # first == null || first.isEmpty()
+      org.apache.commons.lang3.StringUtils # StringUtils.isNotEmpty(first) # first != null && !first.isEmpty()
+      org.apache.maven.shared.utils.StringUtils # StringUtils.isEmpty(first) # first == null || first.isEmpty()
+      org.apache.maven.shared.utils.StringUtils # StringUtils.isNotEmpty(first) # first != null && !first.isEmpty()
+      org.codehaus.plexus.util.StringUtils # StringUtils.isEmpty(first) # first == null || first.isEmpty()
+      org.codehaus.plexus.util.StringUtils # StringUtils.isNotEmpty(first) # first != null && !first.isEmpty()
+      """)
+    void replaceNow(String classname, String beforeLine, String afterLine) {
+        // language=java
+        rewriteRun(
+          java(
+            """
+              import %s;
+
+              class A {
+                  boolean test(String first) {
+                      return %s;
+                  }
+              }
+              """.formatted(classname, beforeLine),
+            """
+              class A {
+                  boolean test(String first) {
+                      return %s;
+                  }
+              }
+              """.formatted(afterLine)));
+    }
+
+    @ParameterizedTest
+    @CsvSource(delimiter = '#', textBlock = """
+      org.apache.commons.lang3.StringUtils # !StringUtils.isEmpty(first) # !(first == null || first.isEmpty())
+      org.apache.commons.lang3.StringUtils # !StringUtils.isNotEmpty(first) # !(first != null && !first.isEmpty())
+      org.apache.commons.lang3.StringUtils # !(StringUtils.isEmpty(first)) # !(first == null || first.isEmpty())
+      org.apache.commons.lang3.StringUtils # !(StringUtils.isNotEmpty(first)) # !(first != null && !first.isEmpty())
+      org.apache.maven.shared.utils.StringUtils # !StringUtils.isEmpty(first) # !(first == null || first.isEmpty())
+      org.apache.maven.shared.utils.StringUtils # !StringUtils.isNotEmpty(first) # !(first != null && !first.isEmpty())
+      org.codehaus.plexus.util.StringUtils # !StringUtils.isEmpty(first) # !(first == null || first.isEmpty())
+      org.codehaus.plexus.util.StringUtils # !StringUtils.isNotEmpty(first) # !(first != null && !first.isEmpty())
```

Comment:
Would have been nicer if we could drop the negation; but `visitUnary` only happens after `visitMethodInvocation`, so if we want to invert the method calls we would need another recipe to run first.

Alternatively we don't yet have a recipe to push down the negation it seems; otherwise that could have been a doAfterVisit.

Reply:
I'd also tried to determine from the method invocation if that's preceded by an exclamation mark, such that we can return the alternative replacement template, but was not able to do so; Any suggestions welcome!

Reply:
I think we should do it the way you first thought: Implement a dedicated visitor which is applied to the unary expression.

Possibly we can come up with some additional patterns we can integrate into `SimplifyBooleanExpression`. E.g. if a binary contains at least one negated unary or another binary with a relational operator. I am just uncertain, because which version is simpler is often debatable and situation dependent.